crypto/evp/kem.c

```diff
@@ -143,6 +144,7 @@ static int evp_kem_init(EVP_PKEY_CTX *ctx, int operation,
 
     if (provkey == NULL) {
         EVP_KEM_free(kem);
+        kem = NULL;
```

Comment:
`kem` is not used in code after `err:`, this assignment is useless.

Reply:
Yes the code is either.. There was an error so free it and exit, else assign it to another object. There is no scope for this changing provided that EVP_KEYMGMT_free() frees its assigned resources